### PR TITLE
Implement recipe instantiation by merging into an active recipe stored in the arc.

### PR DESCRIPTION
**Pull request by mmandlis**

### CODE REVIEW

File: runtime/arc.js [hunk 1]
```diff
@@ -159,7 +169,30 @@ class Arc {
     return s;
   }
 
-  connectParticleToView(particleHandle, particle, name, targetView) {
+  instantiate(recipe) {
+    assert(recipe.isResolved(), 'Cannot instantiate an unresolved recipe');
+    let {views, particles, slots} = recipe.mergeInfo(this._activeRecipe);
+    let recipeViewMap = new Map();
+    views.forEach(recipeView => {
```
Comment (shans) [inline runtime/arc.js]: These views are only the views that need to be created after the merge, 
not all the views referenced by particles in recipe.

File: runtime/arc.js [hunk 2]
```diff
@@ -159,7 +169,30 @@ class Arc {
...
+      let view;
+      if (recipeView.create) {
```
Comment (shans) [inline runtime/arc.js]: so all of these should have the create flag set.

File: runtime/arc.js [hunk 3]
```diff
@@ -159,7 +169,30 @@ class Arc {
...
+        view = this.createView(recipeView.type);
```
Comment (shans) [inline runtime/arc.js]: after creating a view, we also need to copy its ID back into the active recipe and remove the create flag.

File: runtime/arc.js [hunk 4]
```diff
@@ -159,7 +169,30 @@ class Arc {
...
+      } else {
+        view = this.viewById(recipeView.id);
+        assert(view, `view '${recipeView.id}' is not registered in arc`);
+      }
+      recipeViewMap.set(recipeView, view);
+    });
+
+    particles.forEach(recipeParticle => this.instantiateParticle(recipeParticle, recipeViewMap));
```
Comment (shans) [inline runtime/arc.js]: inside this forEach loop you'll need to look up all viewConnections referenced from the particle, extract the recipe views, then use viewById to find the actual views.

File: runtime/arc.js (hunk 3 above)
Comment (shans) [inline runtime/arc.js]: Maybe add a TODO to move the call to OuterPEC's DefineView to here too.

File: runtime/arc.js (hunk 2 above)
Comment (mmandlis) [inline runtime/arc.js]: as discussed offline, i switched to creating the new ones, updating their ids and create flag, and asserting all the rest already exist.

File: runtime/arc.js (hunk 3 above)
Comment (mmandlis) [inline runtime/arc.js]: done, done.

File: runtime/arc.js (hunk 4 above)
Comment (mmandlis) [inline runtime/arc.js]: just realized that i don't need this map at all, can just call viewById in instantiateParticle.

**Comment by mmandlis** [comment]
addressed comments,PTAL

**Pull request merged**